index.bs

```diff
@@ -2661,6 +2688,99 @@ create them does not matter.
  1. Return « |branch1|, |branch2| ».
 </div>
 
+<div algorithm>
+To <dfn lt="fully read">fully read</dfn> a {{ReadableStream}} |stream|, given an algorithm |processBody|, and an algorithm |processBodyError|, run these steps. |processBody| must be an algorithm accepting a [=byte sequence=]. |processBodyError| must be an algorithm optionally accepting an exception.
+
+ 1. Let |successSteps| given a [=byte sequence=] |bytes| be to run |processBody| given |bytes|.
+ 1. Let |errorSteps| optionally given an exception |exception| be to run |processBodyError| given |exception|.
+ 1. Let |reader| be the result of getting a reader for |stream|. If that threw an exception, then run |errorSteps| with that exception and return.
+ 1. [=read all bytes|Read all bytes=] from |reader|, given |successSteps| and |errorSteps|.
+</div>
+
+<div algorithm>
+The <dfn lt="consume fully with abort">consume fully with abort</dfn> algorithm, given a {{ReadableStream}} |stream|, an optional {{AbortSignal}} |signal|, and an algorithm that takes a [=byte sequence=] and returns a JavaScript or throws an exception |convertBytesToJSValue|, runs these steps:
+
+ 1. Let |promise| be [=a new promise=].
+ 1. Let |errorSteps| given |error| to be [=reject=] |promise| with |error|.
+ 1. Let |successSteps| given a [=byte sequence=] |data| be to [=resolve=] |promise| with the result of running |convertBytesToJSValue| with |data|. If that threw an exception, then run |errorSteps| with that exception.
```

Comment:
Perhaps you want to name them as processBody and processBodyError as that's what the parameters of "fully read" are? Or maybe "fully read" parameters should be success/errorSteps instead.